modules/data_pipeline/main.tf

```diff
@@ -1,29 +1,54 @@
 data "google_project" "project" {
 }
 
+locals {
+  source_files = concat(
+    # Add paths to files from /cloud_functions's root
+    [
+      "${path.module}/../../climateiq-cnn/usl_pipeline/cloud_functions/main.py",
+      "${path.module}/../../climateiq-cnn/usl_pipeline/cloud_functions/requirements.txt"
+    ],
+    # Add paths to all .py files from /usl_lib subdirectory
+    tolist(fileset("${path.module}/../../climateiq-cnn/usl_pipeline/usl_lib/usl_lib/", "**/*.py")),
+    # Add paths to all .whl files from /cloud_functions's /wheels subdirectory
+    tolist(fileset("${path.module}/../../climateiq-cnn/usl_pipeline/cloud_functions/wheels/", "*.whl"))
+  )
+
+  usl_lib_dir = "${path.module}/../../climateiq-cnn/usl_pipeline/usl_lib/usl_lib/"
+  wheels_dir = "${path.module}/../../climateiq-cnn/usl_pipeline/cloud_functions/wheels/"
+}
+
+# Read and process the content from source_files
+data "template_file" "t_file" {
+  count = "${length(local.source_files)}"
+  template = element(local.source_files, count.index)
+}
+
+# Copy files to a temporary directory to prep for zip
+resource "local_file" "to_temp_dir" {
+  count    = "${length(local.source_files)}"
+
+  # Filenames are constructed to preserve source subdirectory structure within /temp
+  filename = "${path.module}/temp/${
+    contains(fileset(local.usl_lib_dir, "**/"), element(local.source_files, count.index)) ?        
```

Comment:
might consider making these three separate 'to_temp_dir' resources rather than one with a `contains` switch if you think that'd be more readable